### DISCUSSION TITLE
Can ImageMagick protect against Decompression Bombs?

### DISCUSSION THREAD

**Discussion by MrKomodoDragon** [ImageMagick/ImageMagick#3763] [Help] [answered]
Hi, I’m using wand, a binding for ImageMagick in my program, does image magick have a way to detect if a possible decompression bomb is being used?

**Comment by snibgo** [comment, marked as answer]
Settings in policy.xml guard against files that contain stupidly large numbers of pixels.

**Reply by MrKomodoDragon** [reply]
Thank you!

**Reply by asze17** [reply]
> Settings in policy.xml guard against files that contain stupidly large numbers of pixels.

which exact settings should we change? I am guessing width and height

**Reply by snibgo** [reply]
The settings for width, height and area are directly relevant to the number of pixels. For example, you might want to allow a width and height of 100k pixels, but limit the total area to 100m pixels.

You might also limit the values for memory, map and disk.

After setting values in policy.xml, you can check them with "magick -list resource".

**Comment by fmw42** [comment]
Out of curiosity, why are you so concerned about this type of attack?
